Categories and subcategories

BelisAliosmanova opened the pull request:
CRUD Operations with Categories and Subcategories

 - Log in as an ADMIN to access the admin menu and perform CRUD operations.
 - Navigate through all dropdowns and menus related to categories and subcategories
 - Provide feedback on any aspect you find could be improved or adjusted

If you encounter any issues or have ideas for additional features, please document them clearly.

Also, as you can see, there is a design bug when the URL contains more than one '/'. I couldn't fix it, please try to help.

BelisAliosmanova committed:
create categories and subcategories

BelisAliosmanova committed:
add GlobalControllerAdvice to display all the categories with their subcategories in the navigation bars

BelisAliosmanova committed:
templates for subcategories

BelisAliosmanova committed:
admin only menus

BelisAliosmanova committed:
check session for admin role

stefan-petrov1 reviewed (changes requested):
I get "Access Denied" error whenever I try to perform an admin action even if I am logged as an admin. I checked why that happens and the exception occurs because there are no cookies attached to the request. Our server works with cookies. Send the cookies when requesting. Also I've noticed that the cookies are not recieved when sending a request either.

If this can't be done on the FE - message me and continue with the other tasks.

Other tasks:
   - search in admin menu
   - created_at, updated_at and deleted_at shoud be visible to the admin
   - sort by created_at, updated_at and deleted_at
   - pagination

BelisAliosmanova committed:
data table for categories

BelisAliosmanova committed:
data table for subcategories

stefan-petrov1 reviewed (changes requested):
Whenever I try to delete an entity that already has been deleted I get this error:
<img width="574" alt="image" src="https://github.com/user-attachments/assets/733e9ea5-0c4a-43b0-995e-4cee2e8d08ed">

Also there are no icons for the sorting on the subcategory table:
<img width="1094" alt="image" src="https://github.com/user-attachments/assets/1955d69c-8e2e-42f7-8385-e1432efdbe55">

Also when I try to edit a deleted entity I get this error:
<img width="669" alt="image" src="https://github.com/user-attachments/assets/d0c3d400-8863-4aa3-8529-bf032c4385d1">

The margin is missing on the {n} entires per page at the subcategory admin table:
<img width="1126" alt="image" src="https://github.com/user-attachments/assets/2df9125f-54c4-47ec-97b2-c09c6d0e9554">

Feature idea:
- Make it so the user can toggle between seeing all the active and the inactive entities
- - Active means not deleted and inactive means deleted

A question to you - Should we make it so the deleted entities can be restored if wanted?

BelisAliosmanova committed:
fix bugs and create undo delete

BelisAliosmanova committed:
Add a toddler - the best I can do

If checked (showActive is true), only rows with no deletedAt value (active categories) are shown.
 If unchecked, all rows are displayed.

BelisAliosmanova committed:
entries per page
